app/app/components/Header.jsx

```diff
@@ -1,63 +1,97 @@
-    const router = useRouter(); // Use the router to navigate programmatically
-
-    const handleSearchChange = (e) => {
-        setSearchQuery(e.target.value); // Update the search query
-    };
-
-    // Handle pressing Enter to trigger search
-    const handleSearchKeyPress = (e) => {
-        if (e.key === "Enter" && searchQuery.trim() !== "") {
-            router.push(`/search?q=${encodeURIComponent(searchQuery.trim())}`); // Navigate to the search page with query
-        }
-    };
-
-    return (
-        <header className={styles.header}>
-            <div className={styles.logoContainer}>
-                <Link href="/">
-                    <Image
-                        src="/logo.png"
-                        alt="Leaf Notes Logo"
-                        className={styles.logo}
-                        width={150}
-                        height={50}
-                    />
-                </Link>
-            </div>
-            <nav className={styles.nav}>
-                <ul className={styles.navList}>
-                    <li className={styles.navItem}>
-                        <Link href="/dashboard">DASHBOARD</Link>
-                    </li>
-                    <li className={styles.navItem}>
-                        <Link href="/login">LOG IN</Link>
-                    </li>
-                    <li className={styles.navItem}>
-                        <Link href="/signup">SIGN UP</Link>
-                    </li>{" "}
-                </ul>
-
-                <input
-                    type="text"
-                    placeholder="Search..."
-                    className={styles.searchInput}
-                    value={searchQuery}
-                    onChange={handleSearchChange}
-                    onKeyDown={handleSearchKeyPress}
-                />
-            </nav>
-        </header>
-    );
+  const router = useRouter();
+  const [searchQuery, setSearchQuery] = useState("");
+  const { currentUser, logout } = useAuth(); // Use currentUser and logout from AuthContext
+
+  // Conditional greeting message: Show username if logged in, or 'Guest' if not
+  const userName = useMemo(
+    () => currentUser?.user.firstName || "Guest",
+    [currentUser]
+  );
+
+  // For debugging purposes
+  console.log(`currentUser: ${JSON.stringify(currentUser, null, 2)}`);
+
+  // Handle search input change
+  const handleSearchChange = (e) => {
+    setSearchQuery(e.target.value);
+  };
+
+  // Handle search on "Enter" key press
+  const handleSearchKeyPress = (e) => {
+    if (e.key === "Enter" && searchQuery.trim() !== "") {
+      router.push(`/search?q=${encodeURIComponent(searchQuery.trim())}`);
+    }
+  };
+
+  // Handle logout functionality
+  const handleLogout = async () => {
+    try {
+      await logout(); // Call the logout function from AuthContext
+      router.push("/login"); // Redirect to the login page after logging out
+    } catch (error) {
+      console.error("Logout failed:", error.message);
```

Comment:
usually after a failed action some sort of feedback to the user is necessary like a popup or banner with clear information and call to action